docs/docs/apis/core/plutus-api.md

```diff
@@ -0,0 +1,250 @@
+---
+title: "Plutus API"
+description: "Smart contract interaction with PlutusData, scripts, and blueprints"
+sidebar_position: 5
+---
+
+# Plutus API
+
+The Plutus API provides comprehensive functionality for interacting with Plutus smart contracts on the Cardano blockchain. It includes support for Plutus scripts (V1, V2, V3), data serialization, blueprint loading, and object conversion with type safety and annotation support.
+
+## Key Features
+
+- **Plutus Script Support**: Complete support for V1, V2, and V3 script versions
+- **Data Serialization**: Convert between Java objects and PlutusData
+- **Blueprint Integration**: Load and work with Plutus contract blueprints
+- **Type Safety**: Strong typing for Plutus data structures
+- **Annotation Support**: Automatic conversion using annotations
+- **Object Conversion**: Seamless Java object to PlutusData conversion
+
+## Core Classes
+
+### PlutusScript Classes
+- `PlutusV1Script`: Plutus V1 script implementation
+- `PlutusV2Script`: Plutus V2 script implementation  
+- `PlutusV3Script`: Plutus V3 script implementation
+
+### PlutusData Classes
+- `PlutusData`: Base interface for all Plutus data types
+- `BigIntPlutusData`: Integer data representation
+- `BytesPlutusData`: Byte string data representation
+- `ListPlutusData`: List data representation
+- `MapPlutusData`: Map data representation
+
+### Utility Classes
+- `PlutusObjectConverter`: Convert Java objects to PlutusData
+- `PlutusBlueprintLoader`: Load contract blueprints
+- `PlutusBlueprintUtil`: Blueprint utility functions
+
+## Usage Examples
+
+### Creating Plutus Scripts
+
+Create different versions of Plutus scripts:
+
+```java
+// Create Plutus V2 Script (most common)
+PlutusV2Script v2Script = PlutusV2Script.builder()
+        .type("PlutusScriptV2")
+        .cborHex("590a4d590a4a0100003333...")
+        .build();
+
+// Create Plutus V3 Script (latest)
+PlutusV3Script v3Script = PlutusV3Script.builder()
+        .type("PlutusScriptV3")
+        .cborHex("590b5e590b5b0100003333...")
+        .build();
+
+System.out.println("Scripts created successfully");
+```
+
+### Working with PlutusData
+
+Create and manipulate PlutusData structures:
+
+```java
+// Create integer data
+PlutusData intData = BigIntPlutusData.of(BigInteger.valueOf(42));
+
+// Create string data (as bytes)
+PlutusData stringData = BytesPlutusData.of("Hello Cardano".getBytes());
+
+// Create list data
+List<PlutusData> dataList = Arrays.asList(intData, stringData);
+PlutusData listData = ListPlutusData.of(dataList);
+
+// Create map data
+Map<PlutusData, PlutusData> dataMap = new HashMap<>();
+dataMap.put(BytesPlutusData.of("key".getBytes()), intData);
+PlutusData mapData = MapPlutusData.of(dataMap);
+
+System.out.println("PlutusData structures created");
+```
+
+### Object to PlutusData Conversion
+
+Convert Java objects to PlutusData using annotations:
+
+```java
+@Data
```

Comment:
Runtime annotation processor is deprecated.  I think we can remove `annotation` and `blueprint` related docs from this page. This can be for core plutus api like PlutusScript and PlutusData as you have mentioned.

We can have separate pages for 
- Plutus annotations and build time annotation processor details
- Blueprint support

You can check the existing docs https://cardano-client.dev/docs/annotations/plutus-data-annotations